src/acrcssc/azext_acrcssc/helper/_taskoperations.py

```diff
@@ -99,10 +102,35 @@ def _create_cssc_workflow(cmd, registry, schedule_cron_expression, resource_grou
         dry_run
     )
 
-    logger.warning(f"Deployment of {CONTINUOUS_PATCHING_WORKFLOW_NAME} tasks completed successfully.")
+    if not silent_execution:
+        print(f"Deployment of {CONTINUOUS_PATCHING_WORKFLOW_NAME} tasks completed successfully.")
+
+
+def _update_cssc_workflow(cmd, registry, schedule_cron_expression, resource_group, dry_run, task_list):
+    # compare the task definition to the existing tasks, if there is a difference, we need to update the tasks
+    # if we need to update the tasks, we will update the cron expression from it
+    # if not we just update the cron expression from the given parameter
+    for task in task_list:
+        deployed_task = task.step.encoded_task_content
+        extension_task = _create_encoded_task(CONTINUOUSPATCH_TASK_DEFINITION[task.name]["template_file"])
+        if deployed_task != extension_task:
+            logger.debug(f"Task {task.name} is different from the extension task, updating the task")
+
+            if schedule_cron_expression is None:
+                trigger_task = next((t for t in task_list if t.name == CONTINUOUSPATCH_TASK_SCANREGISTRY_NAME), None)
+                if trigger_task is None:
+                    raise AzCLIError(f"Task {CONTINUOUSPATCH_TASK_SCANREGISTRY_NAME} not found in the registry")
+                if not trigger_task.trigger.timer_triggers:
+                    raise AzCLIError(f"No timer triggers found for task {CONTINUOUSPATCH_TASK_SCANREGISTRY_NAME}")
+                schedule_cron_expression = trigger_task.trigger.timer_triggers[0].schedule
 
+            _create_cssc_workflow(cmd, registry, schedule_cron_expression, resource_group, dry_run, silent_execution=True)
+
+            # the deployment will also update the schedule if it was set, we no longer need to manually set it
+            return
+
+    logger.debug("No difference found between the existing tasks and the extension tasks")
 
-def _update_cssc_workflow(cmd, registry, schedule_cron_expression, resource_group, dry_run):
     if schedule_cron_expression is not None:
         _update_task_schedule(cmd, registry, schedule_cron_expression, resource_group, dry_run)
```

Comment:
Do you still need _update_task_schedule if the task has been updated in line 127?

Reply:
this line is only on the case where no task update is necessary, but a schedule update is specified, in this case the update is done by an SDK call instead of an ARM redeployment.